### ISSUE TITLE
grid embedding vs stream show opposite directions

### ISSUE THREAD

**Issue by fidelram** [theislab/scvelo#72]
Similar to #55 I see a discordance of arrows when using the grid embedding:

The preprocessing had been done using scvelo:

```PYTHON
    scv.pp.filter_and_normalize(adata, n_top_genes=2000)
    scv.pp.pca(adata)
    scv.pp.neighbors(adata)
    scv.pp.moments(adata, n_pcs=15, n_neighbors=30)
    scv.tl.velocity(adata, mode='stochastic')
    scv.tl.velocity_graph(adata, basis='draw_graph_fa")
    scv.tl.velocity_embedding(adata, basis='draw_graph_fa")
```
Then I select a single cluster to look in detail at the arrows:
````PYTHON
adata_cluster = adata[adata.obs.leiden=='10'].copy()
````
![image](https://user-images.githubusercontent.com/4964309/62110540-eeaed680-b2ae-11e9-835a-58be15bc6bba.png)

However, the grid arrows point in the opposite direction (the stream arrows agree with the individual velocities):
![image](https://user-images.githubusercontent.com/4964309/62127159-b1a80b80-b2d1-11e9-82f7-f81a051ed419.png)

To try to address the problem I recompute the velocity embedding:

```PYTHON
    scv.tl.velocity_embedding(adata_cluster, basis='draw_graph_fa")
```
Now the grid arrows agreed with the individual velocity:

![image](https://user-images.githubusercontent.com/4964309/62127297-0055a580-b2d2-11e9-927b-f17ea7e20a11.png)

But if I look at the velocity arrows again, now they changed direction (the stream plot also agrees with the individual velocities):

![image](https://user-images.githubusercontent.com/4964309/62127348-20856480-b2d2-11e9-8295-db60867c265d.png)

**Interestingly** this only happens to some clusters that seem to overlap in 2D with other clusters. For the majority of clusters I don't have a problem.

The results are reproducible also using the syntax:

```PYTHON
scv.tl.velocity_embedding(adata, basis='draw_graph_fa", color='leiden', groups=['10'])
scv.tl.velocity_embedding_grid(adata, basis='draw_graph_fa", color='leiden', groups=['10'])
```

Could you help me find the reason behind this? It is very annoying to see that the arrows change direction.

**Comment by VolkerBergen** [comment]
Thanks for raising this. I suspect that there is an issue when subsetting your data, e.g. due to a distorted neighbor graph. I will in investing this; guess that having this observed in overlapping clusters might be a good hint to follow up.

**Comment by VolkerBergen** [comment]
I found the bug. Thousand thanks for leading me there! 

This is what happened: We implemented an autoscale module for the arrows/quiver in the grid embedding. The rescaling factor computed in https://github.com/theislab/scvelo/blob/3ea66025ec6b1e2bde54643c001a94d744f1546d/scvelo/tools/velocity_embedding.py#L13 depends on the maximum value of your embedding coordinates. Now, subsetting anndata and thus subsetting the embedding, this maximum value can become negative, thus changing the sign of the velocities.

I solved this and will include a warning that this issue may have occurred when subsetting anndata.
